src/tss2-tcti/tcti-device.c

```diff
@@ -131,9 +131,13 @@ tcti_device_receive (
                      *response_size, tcti_intel->header.size);
         return TSS2_TCTI_RC_INSUFFICIENT_BUFFER;
     }
+    rc = header_marshal (&tcti_intel->header, response_buffer);
```

Comment:
Do we need the header_buf at all? We could just read the header into to the response_buffer and 
get the header from it: parse_header (response_buffer, &tcti_intel->header);
This way we would not need the header_marshal at all.

Reply:
Yes, because an application may call tcti_receive() with buffer==NULL.

Reply:
When this code was being refactored I tried both of these approaches. Unmarshalling and storing the header this way just looked better than unmarshalling the size field from a buffer when we needed it. If we're looking to be more efficient we could get away with using 6 instead of 10 bytes since we don't do anything with the RC but I didn't see any harm in storing the whole header.